File: app/src/main/java/com/catherine/webservices/toolkits/CLog.java
package com.catherine.webservices.toolkits;

import android.util.Log;

import com.catherine.webservices.BuildConfig;

/**
 * Created by Catherine on 2017/7/17.
 * Soft-World Inc.
 * catherine919@soft-world.com.tw
 */

public class CLog {
    private static final boolean DEBUG = BuildConfig.SHOW_LOG;

    /**
     * Not work while proguard enabled
     *
     * @return class simple name
     */
    public static String getTag() {
        String tag = "";
        final StackTraceElement[] ste = Thread.currentThread().getStackTrace();
        for (int i = 0; i < ste.length; i++) {
            if (ste[i].getMethodName().equals("getTag")) {
                tag = "(" + ste[i + 1].getFileName() + ":" + ste[i + 1].getLineNumber() + ")";
            }
        }
        return tag;
    }

    public static void v(String tab, String message) {
        if (DEBUG) {
            Log.v(tab, message);
        }
    }

    public static void d(String tab, String message) {
        if (DEBUG) {
            Log.d(tab, message);
        }
    }

    public static void e(String tab, String message) {
        if (DEBUG) {
            Log.e(tab, message);
        }
    }

    public static void w(String tab, String message) {
        if (DEBUG) {
            Log.w(tab, message);
        }
    }

    public static void i(String tab, String message) {
        if (DEBUG) {
            Log.i(tab, message);
        }
    }

    public static void v(String message) {
        if (DEBUG) {
            Log.v(getTag(), message);
        }
    }

    public static void d(String message) {
        if (DEBUG) {
            Log.d(getTag(), message);
        }
    }

    public static void e(String message) {
        if (DEBUG) {
            Log.e(getTag(), message);
        }
    }

    public static void w(String message) {
        if (DEBUG) {
            Log.w(getTag(), message);
        }
    }

    public static void i(String message) {
        if (DEBUG) {
            Log.i(getTag(), message);
        }
    }
}

File: app/src/main/java/com/catherine/webservices/toolkits/CLog.kt
package com.catherine.webservices.toolkits

import android.util.Log

import com.catherine.webservices.BuildConfig

/**
 * Created by Catherine on 2017/7/17.
 * Soft-World Inc.
 * catherine919@soft-world.com.tw
 */

internal class CLog {

    companion object {
        val DEBUG = BuildConfig.SHOW_LOG

        fun getTag(): String {
            var tag = ""
            val ste = Thread.currentThread().stackTrace
            for (i in ste.indices) {
                if (ste[i].methodName == "getTag") {
                    tag = "(${ste[i + 1].fileName}:${ste[i + 1].lineNumber})"
                }
            }
            return tag
        }

        fun v(tag: String = "Default TAG", message: String = "Default message") {
            if (DEBUG)
                Log.v(tag, message)
        }

        fun d(tag: String = "Default TAG", message: String = "Default message") {
            if (DEBUG)
                Log.d(tag, message)
        }

        fun i(tag: String = "Default TAG", message: String = "Default message") {
            if (DEBUG)
                Log.i(tag, message)
        }

        fun w(tag: String = "Default TAG", message: String = "Default message") {
            if (DEBUG)
                Log.w(tag, message)
        }

        fun e(tag: String = "Default TAG", message: String = "Default message") {
            if (DEBUG)
                Log.e(tag, message)
        }

        fun v(message: String = "Default message") {
            if (DEBUG)
                Log.v(getTag(), message)
        }

        fun d(message: String = "Default message") {
            if (DEBUG)
                Log.d(getTag(), message)
        }

        fun i(message: String = "Default message") {
            if (DEBUG)
                Log.i(getTag(), message)
        }

        fun w(message: String = "Default message") {
            if (DEBUG)
                Log.w(getTag(), message)
        }

        fun e(message: String = "Default message") {
            if (DEBUG)
                Log.e(getTag(), message)
        }
    }


//    final StackTraceElement[] ste = Thread.currentThread().getStackTrace();
//    for (int i = 0; i < ste.length; i++) {
//        if (ste[i].getMethodName().equals("getTag")) {
//            tag = "(" + ste[i + 1].getFileName() + ":" + ste[i + 1].getLineNumber() + ")";
//        }
//    }
}